docs/src/snap/howto/cis-hardening.md

```diff
@@ -282,6 +347,3230 @@ the sets, including the dqlite specific checks in the output.
+```
+
+##### Control 1.2.8
+
+**Description:**
+
+Ensure that the --authorization-mode argument includes RBAC
+(Automated)
+
+
+**Remediation:**
+
+Edit the API server configuration file /var/snap/k8s/common/args/kube-apiserver
+on the control plane node and set the --authorization-mode
+parameter to a value that includes RBAC,
+
+`--authorization-mode=Node,RBAC`
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-apiserver | grep -v grep
+```
+
+**Expected output:**
+
+```
+--authorization-mode=Node,RBAC
+```
+
+##### Control 1.2.9
+
+**Description:**
+
+Ensure that the admission control plugin EventRateLimit is
+set (Manual)
+
+
+**Remediation:**
+
+Follow the Kubernetes documentation and set the desired
+limits in a configuration file.
+Then, edit the API server configuration file
+/var/snap/k8s/common/args/kube-apiserver and set the following arguments.
+
+```
+--enable-admission-plugins=...,EventRateLimit,...
+--admission-control-config-file=<path/to/configuration/file>
+```
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-apiserver | grep -v grep
+```
+
+**Expected output:**
+
+```
+--enable-admission-
+plugins=NodeRestriction,EventRateLimit,AlwaysPullImages
+```
+
+##### Control 1.2.10
+
+**Description:**
+
+Ensure that the admission control plugin AlwaysAdmit is not
+set (Automated)
+
+
+**Remediation:**
+
+Edit the API server configuration file /var/snap/k8s/common/args/kube-apiserver
+on the control plane node and either remove the --enable-
+admission-plugins parameter, or set it to a
+value that does not include AlwaysAdmit.
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-apiserver | grep -v grep
+```
+
+**Expected output:**
+
+```
+--enable-admission-
+plugins=NodeRestriction,EventRateLimit,AlwaysPullImages
+```
+
+##### Control 1.2.11
+
+**Description:**
+
+Ensure that the admission control plugin AlwaysPullImages is
+set (Manual)
+
+
+**Remediation:**
+
+Edit the API server configuration file /var/snap/k8s/common/args/kube-apiserver
+on the control plane node and set the --enable-admission-
+plugins parameter to include
+AlwaysPullImages.
+
+`--enable-admission-plugins=...,AlwaysPullImages,...`
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-apiserver | grep -v grep
+```
+
+**Expected output:**
+
+```
+--enable-admission-
+plugins=NodeRestriction,EventRateLimit,AlwaysPullImages
+```
+
+##### Control 1.2.12
+
+**Description:**
+
+Ensure that the admission control plugin SecurityContextDeny
+is set if PodSecurityPolicy is not used (Manual)
+
+
+**Remediation:**
+
+Edit the API server configuration file /var/snap/k8s/common/args/kube-apiserver
+on the control plane node and set the --enable-admission-
+plugins parameter to include
+SecurityContextDeny, unless PodSecurityPolicy is already in
+place.
+
+`--enable-admission-plugins=...,SecurityContextDeny,...`
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-apiserver | grep -v grep
+```
+
+**Expected output:**
+
+```
+--enable-admission-
```

Comment:
expected output does not include securityContextDeny. Is this the expected output before any editing to the file? Is podSecurityPolicy in place so we don't need it?

Reply:
This is a case where the CIS recommendation suggests something that is outdated, https://kubernetes.io/docs/tasks/configure-pod-container/migrate-from-psp/ the check is manual so each admin can select what matches his needs best.